Station Pets - Loadouts, Objectives, Wawa!

FairlySadPanda opened the pull request:
![image](https://github.com/user-attachments/assets/64044fa3-9c18-4d7e-a0df-25a58ade5b46)

This design document outlines a proposal to lift the pet freeze by moving pets over to becoming loadout items, unlocked via playtime for various job roles. In addition, it outlines gameplay mechanics that encourage players to roleplay owning and taking care of their pets. Afterwards, it outlines the principles behind pet ghost roles and when a pet should be unlocked for latejoiners and the recently deceased to play. Pet Carriers are discussed, and we then dwell for a short time on the Captain and Head Of Personnel's pets and their gameplay importance. Finally, it covers the unique trio of pet-crew ancestors (Pun Pun, Kweh Kweh and Wa Wa), their purpose in gameplay, and their unique circumstances as Junior Assistant Bartenders.

This document provides a vector for introducing scurrets to the game ahead of their starring role as the protagonists of the Pizza Delivery Critter role. This is done via introducing them via Wa Wa, an unlockable Bartender pet, alongside Kweh Kweh, a kobold bartender.

FairlySadPanda committed:
Station Pets - Loadouts, Objectives, Wawa!

FairlySadPanda commented:
Something something "chipping" station pets with something so AI can see  them (think mindshields)

walksanatora commented:
I like the ability to select stataion pets roun-start. mabey command will actually start taking care of (evacuating) with their pets.

FairlySadPanda committed:
Add remaining needed notes, prototyped Strike Team proposal document

FairlySadPanda committed:
Linked the proposal

FairlySadPanda commented:
I've updated this document to cover the workload I'd propose for Strike Team Wawa, incorporate Nik's feedback (with slight editing to make the admin design fit the document's style) and added notes on future workload.

This introduces the notion of "User Stories" to the project, which I am sure is going to horrify certain professional members of this project.

slarticodefast commented on src/en/space-station-14/round-flow/proposals/pets-and-pet-protection.md:
```diff
@@ -0,0 +1,309 @@
+
+# Pet Carriers
+
+If a pet spawner is mapped on a station, that spawner must always be accompanied by a pet carrier.
```
Wait, I thought they are explicitly not mapped, but become part of the loadout?

FairlySadPanda replied:
Yeah, you could just make anyone with an allocated pet spawn holding the carrier.

slarticodefast commented on src/en/space-station-14/round-flow/proposals/pets-and-pet-protection.md:
```diff
@@ -0,0 +1,309 @@
+
+# Hamlet And Friends
+
+The Captain's pet is Mandatory. The Captain must choose from one of the following options at the start of the round.
```
Probably for the best if the captain only has one pet instead of two.
But the captain being able to select if there are extra TC on the station or not leads into powergaming territory.

FairlySadPanda replied:
True, I'd welcome suggestions on balance as I don't know here. You could potentially have the Captain's pet always contain TC, but then the Cap has to choose between "safe docile fox" and "anger and hunger given hamster form" and that means you will never see Hamlet.

I think having the Captain have to decide if they want to have a traitor goal or a shitter ghost role is a good tradeoff but I'd welcome help on this.

FairlySadPanda replied:
@slarticodefast I've updated this design. Let me know what you think.

slarticodefast commented on src/en/space-station-14/round-flow/proposals/pets-and-pet-protection.md:
```diff
@@ -0,0 +1,309 @@
+
+Pets, outside the Junior Assistant Bartender, are assumed to be creatures that lack hands. They _may_ be capable of dragging things, or even wearing a small container for moving items around. However, ancestor species always have one "hand", reflecting their diminished size. Ancestor species are a significant power-creep over normal animals, and should be avoided as station pets without being bac ...
+
+# Proposed Workload
+
+| Strike Team      | Kanban Organizer | Members |
+| ---------------- | ---------------- | ------- |
+| Strike Team Wawa | FairlySadPanda   | TBD     |
+
+(*Pending acceptance of Strike Team methodology)
+
+(** The following is a demonstration of a work-plan for a development project, with features described via User Stories)
```
if you use the new doc template, that one has a section for "coders" which you can use for this

slarticodefast commented on src/en/space-station-14/round-flow/proposals/pets-and-pet-protection.md:
```diff
@@ -0,0 +1,309 @@
+| Lobby Player | Select a choice of Mandatory Pets for any given job                                                                                    | I can have my preference of pet if mandated by NanoTrasen.                     |
+| Designer     | Allocate Station Pets to be unlocked in Loadouts at a variety of playtimes                                                             | I can reward long-term players with nice unlocks                               |
+| Round Player | Spawn with my selected Optional or Mandatory Pet at round-start *if I was the most senior member of my role at the time*               | I can feel rewarded for my play-time investment.                               |
+| Designer     | Spawn each Mandatory Pet at round-start even without anyone playing that role, using random chance and weights                         | Mandatory pets always start on the station to enforce other gameplay mechanics |
```
what is the designer supposed to be here?
this part is a little confusing

FairlySadPanda replied:
Designers are users of the team output; they need to be able to use this feature to design forward from, so thus should be accounted for in the stories.

slarticodefast commented on src/en/space-station-14/round-flow/proposals/pets-and-pet-protection.md:
```diff
@@ -0,0 +1,309 @@
+| Late Joiner                | Select Junior Assistant Bartender from the late-join menu | I don't need to enter a ghost role raffle if I want to play as a Junior Assistant Bartender |
+| Junior Assistant Bartender | Play as Pun Pun, the Monkey                               | I can OOK and EEE with the best of them, and be a good monkey                               |
+| Junior Assistant Bartender | Play as Kweh Kweh, the Kobold                             | I can KEWH with the best of them, and be a little goober kobold                             |
+| Designer                   | Spawn scurrets                                            | I can use them for unusual ancestor job roles, such as the Pizza Delivery Critter           |
```
same
spawning mobs is more something that admins would do

slarticodefast commented on src/en/space-station-14/round-flow/proposals/pets-and-pet-protection.md:
```diff
@@ -0,0 +1,309 @@
+
+(** The following is a demonstration of a work-plan for a development project, with features described via User Stories)
+
+## Station Pets Selectable as Loadouts
```
I would probably remove this giant table since it just summarizes the above text again

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/pets-and-pet-protection.md:
```diff
@@ -0,0 +1,309 @@
+2. The Head of Personnel's pet. See "Ian And Friends" below.
+3. The Junior Assistant Bartender. See "Junior Assistant Bartenders" below.
+
+Mandatory pets always have a spawn chance weight.
```
Will do

The design is "a mandatory pet must spawn, so as a fallback one is picked randomly" so getting that is trickky

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/pets-and-pet-protection.md:
```diff
@@ -0,0 +1,309 @@
+2. If the player with priority has selected a pet, that pet is spawned and this workflow ends. Otherwise, priority is passed to the player with the next-highest time as that job role and this step is repeated. If no other player can be passed priority, skip to step 3.
+3. The spawn chance weights of each of the Mandatory pet choices for that role are used to randomly select a pet. This pet is spawned.
+
+For example, if no bartender has spawned, Pun's weight might be 0.5, Kweh's weight 0.3 and Wa's rate 0.1. This would give Pun a 5/9 chance of being spawned, and Wa only a 1/9 chance.
```
All Mandatory pets start on station - Renault/Hamlet on Bridge, Pun in their room, the corgi in HOP office.

FairlySadPanda committed:
Update src/en/space-station-14/round-flow/proposals/pets-and-pet-protection.md

Co-authored-by: slarticodefast <161409025+slarticodefast@users.noreply.github.com>

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/pets-and-pet-protection.md:
```diff
@@ -0,0 +1,309 @@
+
+When a pet is spawned due to a player spawning with that job role, that player is given an objective in their mind to ensure that the pet escapes to CentComm alive.
+
+This role should always match the equivalent "escape to CentComm alive" rules that antagonists use.
```
The ONLY antagonist that has pet responsibilites after appointment is the Thief.

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/pets-and-pet-protection.md:
```diff
@@ -0,0 +1,309 @@
+
+For safety, a player may choose to load their pet into a pet carrier, buckle them to a bed or a seat, or any other method to prevent their pet from jumping out the escape shuttle mid-flight. As long as the pet is alive when arriving at CentComm, the player will green-text.
+
+The exception to this is if the pet has fulfilled a steal objective for an antag, such as the Thief or Traitor. For these antagonists, they must directly be holding the pet in their hands, a pet carrier, or other storage on their person.
```
Yeah is a tricky one to solve, to be fair it's the thief's own fault for falling over

PotentiallyTom commented on src/en/space-station-14/round-flow/proposals/pets-and-pet-protection.md:
```diff
@@ -0,0 +1,309 @@
+* Morty, Morticia (QM)
+* Paperwork (Librarian)
+* The Moproach (Janitor)
+* Shiva, McGruff (The HOS)
```
McGriff is usually treated as the warden's pet rather than the HoS's

FairlySadPanda replied:
Most departments will have about one pet - I should make this clearer

So Shiva, McGruff etc go into the pool

FairlySadPanda committed:
Updated, wawa.

FairlySadPanda commented:
Updated the document.

Note that I've done a pass on "the Captain's pet" and the logic for spawning pets.

Of particular note:

1. The Captain's pet is _always_ a ghost role and always has 5 TC on them. Do you wanna play as a fox? You can play as a fox.
1a. This may be the closest @ScarKy0 will be able to play Vulps for the time being ;pensive:
2. Pets _always_ spawn in a Pet Carrier. This simplifies logic and completely removes Optional pets from mapping requirements!
3. A random member of a given role gets a pet, not the most-senior. This corresponds to Bhijn's (fully correct) goal of making SS14 a healthier game for people with obsessive or addictive personalities.

FairlySadPanda commented:
Oh and I've clarified a few free agent and antag interactions - zombies, Para Clones, etc.

FairlySadPanda commented on src/en/space-station-14/round-flow/proposals/pets-and-pet-protection.md:
```diff
@@ -0,0 +1,248 @@
+| Pet                       | Why                                                                                                                        |
+| ------------------------- | -------------------------------------------------------------------------------------------------------------------------- |
+| Hamlet                    | He can run under doors, be places they shouldn't, and eats everything on the station before being shoved into a microwave. |
+| Renault                   | He's more durable than Hamlet but can't move around the station easily. A guard-fox for the bridge.                        |
```
...yes it's a vixen, I forgot

FairlySadPanda committed:
trans

ASTRALIXIX commented:
I think it would be fun to watch Ian as a ghost role too, with his own goals (Of which, of course, would be to stay alive on this station). After all, he's almost the mascot of the space station

sudobeans commented on src/en/space-station-14/round-flow/proposals/pets-and-pet-protection.md:
```diff
@@ -0,0 +1,248 @@
+
+# Pets As Loadouts
+
+Pets are not mapped directly onto the station. Instead, each role may have at least one pet assigned to it via its loadout. Much like other loadout options, pets are unlocked via playtime.
```
How much playtime do you have in mind for this? I recall that Slarticodefast said "Players should not be encouraged to 'grind' the game to an unhealthy degree" in https://github.com/space-wizards/docs/pull/436#discussion_r2044244689

FairlySadPanda replied:
That's a maintainer balance topic rather than something I can express. "In line with maintainer expectations and Bhijn & Myr's overall health goals". 

I personally love big numbers to grind for, but I know that that runs contrary to what Wizden wants to do.

sudobeans reviewed (approved):
This proposal is great. I like the ideas for how antags should treat their pets. I love the pet carrier idea, this could make it easier to evacuate the station with pets.

FairlySadPanda commented:
Just to break down this feedback a bit, I think you're correct in that pets would be frustrating to work with without some way of controlling them. It's important to remember two things:

1. Whilst greentexting is a little pat-on-the-back to players, greentexting is firmly optional
2. It's hard to design ahead at the moment to cope with how players would interact with the system

I'm thus not going to add this "pet follow" mechanic to the doc, but it's definitely something I'd look into during implementation, especially as we already have an example in the Ratking's mice for how it can work in gameplay.

FairlySadPanda commented:
> I think it would be fun to watch Ian as a ghost role too, with his own goals (Of which, of course, would be to stay alive on this station). After all, he's almost the mascot of the space station

For specific pets beyond Pun, Kweh and Wa I'll leave that space open to others. I have seen adorable "playable Ian" sprites on SS13 servers.

Sparlight has been working on the Borgis, who will cover "I wanna be a goddamn corgi and be ADORABLE" pretty well :>

FairlySadPanda commented:
This document has been open for about six weeks and I'm thinking the comments are generally positive, and maintainers do not have big concerns. In addition, the good work that the maintainers have been doing keeping the PR count on Wizden stable tells me that it is currently a good time to work on content.

I am probably going to sit down and try and make this work as a PR; I'm going to focus on Pun, Kweh and Wa first, because bar needs the most help.

FairlySadPanda committed:
Update pets-and-pet-protection.md

FairlySadPanda committed:
Update pets-and-pet-protection.md

Princess-Cheeseballs commented:
Design doc meeting discussed this PR and decided to close it rather than merge. Overall there were concerns about making pets into loadouts implementation wise and interest wise. We'd rather have pets be more individually interesting, like how polly is now, versus trying to make them something the crew is obligated to protect based on job. Overall if we wanted random pets for bartender it'd be best to just use entitytables the same way Ian currently does. If we wanted more interesting pets, we should make them more interesting as NPCs. And if we wanted crew to protect their pets, it should be tied to an overall crew objective system rather than something that is tied to a specific job or loadout choice.
